### PR TITLE
swap review updates

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes RNBW-2653

## PoW (screenshots / screen recordings)
https://cloud.skylarbarrera.com/Screen-Recording-2022-04-20-15-44-53.mp4

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/exchange/ExchangeDetailsRow.js [hunk 1]
```diff
@@ -105,14 +102,6 @@ export default function ExchangeDetailsRow({
     priceImpactScale,
   ]);
 
-  const isWrapEth =
```
Comment (skylarbarrera) [inline src/components/exchange/ExchangeDetailsRow.js:108]: note to check on this logic, dont know why I removed

File: src/components/expanded-state/swap-details/SwapDetailsContent.js [hunk 2]
```diff
@@ -57,104 +50,79 @@ export default function SwapDetailsContent({
     (!isHighPriceImpact || priceImpactNativeAmount) &&
     priceImpactPercentDisplay;
 
-  const { nativeCurrencySymbol } = useAccountSettings();
-
-  const feeNativeCurrency = useMemo(() => {
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsContent.js:62]: I moved this stuff to a hook so its more reusable

File: src/components/expanded-state/swap-details/SwapDetailsContractRow.js [hunk 3]
```diff
@@ -55,16 +70,14 @@ function SwapDetailsContractRowContent({
   scaleTo = 1.06,
   ...props
 }) {
-  const { colors } = useTheme();
+  const defaultColor = useForegroundColor('secondary');
+
   const colorForAsset = useColorForAsset(asset);
-  const animation = useTimingTransition(menuVisible, { duration: 150 });
+  const animation = useTimingTransition(menuVisible, { duration: 80 });
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsContractRow.js:76]: I noticed this animation was pretty glitchy so I smoothed it out

File: src/components/expanded-state/swap-details/SwapDetailsExchangeRow.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import { ImgixImage } from '@rainbow-me/images';
+
+const UniswapUri = 'https://cloud.skylarbarrera.com/Oval.png';
+
+const ExchangeIcon = ({ index, protocol }) => {
+  const { colors } = useTheme();
+  return (
+    <ImgixImage
+      size={22}
+      source={{ uri: UniswapUri }}
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsExchangeRow.js:26]: I need to add all the images to the assets repo still but dont want to block

File: src/components/expanded-state/swap-details/SwapDetailsFeeRow.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import Routes from '@rainbow-me/routes';
+
+export default function SwapDetailsUniswapRow(tradeDetails) {
+  const { navigate } = useNavigation();
+  const { nativeCurrencySymbol } = useAccountSettings();
+  const { rainbowFeeNative, rainbowFeePercentage } = useRainbowFee(
+    tradeDetails
+  );
+
+  //TODO: this isnt i18n friendly
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsFeeRow.js:16]: pinged Marco to make sure this is handled properly

File: src/components/expanded-state/swap-details/SwapDetailsMasthead.js [hunk 6]
```diff
@@ -41,14 +56,29 @@ export default function SwapDetailsMasthead({
         priceValue={inputPriceValue}
         type="input"
       />
-      <Icon color={colors.dark} name="doubleChevron" />
+      <Bleed shadow="21px heavy" space="15px">
+        <Box
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsMasthead.js:60]: should be able to touch up and rm inline styles

File: src/components/expanded-state/swap-details/SwapDetailsPriceRow.js [hunk 7]
```diff
@@ -32,24 +32,23 @@ export default function SwapDetailsPriceRow({ tradeDetails, ...props }) {
 
   const steps = useMemo(
     () => [
-      `${inputExecutionRate} ${outputCurrency?.symbol} per ${inputCurrency?.symbol}`,
-      `${outputExecutionRate} ${inputCurrency?.symbol} per ${outputCurrency?.symbol}`,
+      `1 ${outputCurrency?.symbol} for ${outputExecutionRate} ${inputCurrency?.symbol}`,
```
Comment (skylarbarrera) [inline src/components/expanded-state/swap-details/SwapDetailsPriceRow.js:35]: i18n

File: src/components/icons/svg/DoubleChevronIcon.js [hunk 8]
```diff
@@ -7,14 +7,14 @@ const DoubleChevronIcon = ({ colors, color = colors.black, ...props }) => (
     <Path
       d="M14.111 11.95c0-.42-.146-.762-.488-1.085l-7.49-7.324a1.252 1.252 0 00-.947-.39c-.752 0-1.368.605-1.368 1.357 0 .37.157.713.43.986l6.64 6.446-6.64 6.464c-.273.264-.43.606-.43.987 0 .752.616 1.357 1.368 1.357.37 0 .693-.127.947-.39l7.49-7.325c.342-.322.488-.674.488-1.084z"
       fill={color}
-      fillOpacity="0.3"
```
Comment (skylarbarrera) [inline src/components/icons/svg/DoubleChevronIcon.js:10]: this is not used anywhere anymore, should remove the file

File: src/screens/ExchangeModal.js [hunk 9]
```diff
@@ -32,8 +32,10 @@ import {
 import { FloatingPanel } from '../components/floating-panels';
 import { GasSpeedButton } from '../components/gas';
 import { Centered, KeyboardFixedOpenLayout } from '../components/layout';
-import { FLASHBOTS } from '../config/experimental';
-import useExperimentalFlag from '../config/experimentalHooks';
+/*
+  import { FLASHBOTS } from '../config/experimental';
```
Comment (skylarbarrera) [inline src/screens/ExchangeModal.js:36]: keep this cause its tbd what we're gonna be doing regarding flashbots

File: src/redux/settings.ts [hunk 10]
```diff
@@ -157,6 +171,16 @@ export const settingsChangeTestnetsEnabled = (testnetsEnabled: any) => async (
   saveTestnetsEnabled(testnetsEnabled);
 };
 
+export const settingsChangeFlashbotsEnabled = (flashbotsEnabled: any) => async (
```
Comment (brunobar79) [inline src/redux/settings.ts:174]: flashbotsEnabled should be a boolean, not any

File: src/redux/swap.ts [hunk 11]
```diff
@@ -200,7 +200,7 @@ const INITIAL_STATE: SwapState = {
   independentValue: null,
   inputCurrency: null,
   outputCurrency: null,
-  slippageInBips: 50,
+  slippageInBips: 1000,
```
Comment (brunobar79) [inline src/redux/swap.ts:203]: 10% slippage - wilding. Set it to 1%

File: src/screens/ExchangeModal.js (hunk 9 above)
Comment (brunobar79) [reply src/screens/ExchangeModal.js:36]: localstorage is fine so I'm ok to remove it

File: src/screens/ExchangeModal.js [hunk 12]
```diff
@@ -500,6 +506,40 @@ export default function ExchangeModal({
     ]
   );
 
+  const navigateToSwapSettingsSheet = useCallback(() => {
+    android && Keyboard.dismiss();
+    const lastFocusedInputHandleTemporary = lastFocusedInputHandle.current;
+    android && (lastFocusedInputHandle.current = null);
+    inputFieldRef?.current?.blur();
+    outputFieldRef?.current?.blur();
+    nativeFieldRef?.current?.blur();
```
Comment (brunobar79) [inline src/screens/ExchangeModal.js:515]: Isn't blurring one enough?

**Review by brunobar79 — APPROVED** [review]
🤝

File: src/screens/ExchangeModal.js (hunk 12 above)
Comment (skylarbarrera) [reply src/screens/ExchangeModal.js:515]: prolly but im just following the other instances

File: src/redux/swap.ts (hunk 11 above)
Comment (skylarbarrera) [reply src/redux/swap.ts:203]: im nuts

File: src/components/expanded-state/swap-details/SwapDetailsFeeRow.js (hunk 5 above)
Comment (skylarbarrera) [reply src/components/expanded-state/swap-details/SwapDetailsFeeRow.js:16]: saving this for l8r cause we got some other BS that needs to be done

**Pull request merged**